Assets/Scripts/Models/Trade/Currency.cs

```diff
@@ -8,13 +8,31 @@
 #endregion
 
 using System.Xml;
+using System;
 
 public class Currency
 {
     public string Name;
     public string ShortName;
+    
     public float Balance;
```

Comment:
private float balance;

public float Balance{
get{return balance;}
set{ 
if(balance == value) {return;}
balance = value;
BalanceChangedInvoke(this);
}